otelcollector/deploy/addon-chart/azure-monitor-metrics-addon/templates/ama-metrics-deployment.yaml

```diff
@@ -279,6 +279,26 @@ spec:
                   - key: kubernetes.azure.com/cluster
                     operator: Exists
                   {{- end }}
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                - key: rsName
+                  operator: In
+                  values:
+                  - ama-metrics
+              topologyKey: topology.kubernetes.io/zone
```

Comment:
Not sure what is the effect of this. Doesn't this mean that it will be scheduled only on nodes with k8s.io/zone label ? why is this needed ? same for ds k8s.io/hostname ...

Reply:
if there are more than 1 pods in our deployment, it will try spread to more than 1 AZ (if available) and also if there are more than 1 hosts in the same zone it will spread >1 pods to different hosts . All these automatically happens in AKS from cloud ctrl manager. we still need to set per workload/addon ?

Reply:
@vishiy - This is in the addon chart guideline for deployments with more than 1 replicas - https://msazure.visualstudio.com/CloudNativeCompute/_wiki/wikis/CloudNativeCompute.wiki/386173/Addon-Chart-Yaml-Guideline?anchor=4.--pod-anti-affinity%3A
Basically this is a soft check to see if the pods can be deployed on diff zones and on diff zones

Reply:
Hi Rashmi - Thats the default behavior for AKS when AZ is enabled (instead of every deployment/replicas opting into it to leverage AZ/ single AZ + multiple nodes). Please see here -- https://learn.microsoft.com/en-us/azure/aks/availability-zones#verify-pod-distribution-across-zones. I will ask in the other PR as well so AKS can comment as needed ....

Also will this impact our label based forced scheduling on a stronger/better node pool as well ?

Reply:
Thanks Vishwa. I added this since it came up in the HPA design review discussion with the AKS team the other day and they pointed me to this doc. 
It will not take higher priority over the label based scheduling(weight= 50) on node since that as a higher weight than the podantiaffinity(weight = 10 & 5)